rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -445,8 +445,7 @@ def parse_value_string(type_, value_string):
     if type_.is_primitive_type() and not type_.is_array:
         return parse_primitive_value_string(type_, value_string)
 
-    # TODO(mikaelarguedas) change this condition once string escape function is implemented
-    if type_.is_primitive_type() and type_.is_array and type_.type != 'string':
+    if type_.is_primitive_type() and type_.is_array:
```

Comment:
I don't think allowing single quotes is unnecessarily burdensome, so I'm fine with that, I was just trying to find ways to keep is simple. But not requiring the quotes seems unnecessary to me. For ROS 1, we use yaml syntax on the command line (which worked really well and allowed us to only specify parts of the message). So, if we use that again, I don't think we need a super flexible parser that can work here and on any arbitrary input from the CLI.